STL_Extension/include/CGAL/exceptions.h

```diff
@@ -185,7 +185,19 @@ class Warning_exception : public Failure_exception {
                              "warning condition failed") {}
 };
 
+namespace internal {
 
+// The following classes are useful to create output iterators (with the help
+// of boost::function_output_iterator) that will throw as soon as something is being written.
+class Throw_at_output_exception : public std::exception { };
```

Comment:
That seems a waste to derive from `std::exception`. Why not have an empty exception class?